File: frontend/src/api/sessions.ts
import client from './client'

export interface ActiveSession {
  radacctid: number
  acctsessionid: string
  username: string
  nasipaddress: string
  nasportid: string | null
  calledstationid: string | null
  callingstationid: string | null
  framedipaddress: string | null
  acctstarttime: string | null
  duration_seconds: number
}

export async function getActiveSessions(): Promise<ActiveSession[]> {
  const { data } = await client.get('/sessions/active')
  return data
}

export async function disconnectSession(radacctid: number): Promise<void> {
  await client.delete(`/sessions/${radacctid}`)
}

File: backend/models/sessions.py
from typing import Optional
from pydantic import BaseModel


class ActiveSession(BaseModel):
    radacctid: int
    acctsessionid: str
    username: str
    nasipaddress: str
    nasportid: Optional[str] = None
    calledstationid: Optional[str] = None
    callingstationid: Optional[str] = None
    framedipaddress: Optional[str] = None
    acctstarttime: Optional[str] = None
    duration_seconds: int = 0


class AccountingRecord(BaseModel):
    radacctid: int
    acctsessionid: str
    username: str
    realm: Optional[str] = None
    nasipaddress: str
    calledstationid: Optional[str] = None
    callingstationid: Optional[str] = None
    framedipaddress: Optional[str] = None
    acctstarttime: Optional[str] = None
    acctstoptime: Optional[str] = None
    acctsessiontime: Optional[int] = None
    acctinputoctets: Optional[int] = None
    acctoutputoctets: Optional[int] = None
    acctterminatecause: Optional[str] = None


class AccountingListResponse(BaseModel):
    total: int
    page: int
    per_page: int
    items: list[AccountingRecord]
